src/state/zustand/swapperStore/utils.ts

```diff
@@ -1,8 +1,44 @@
 // Helper function to convert basis points to percentage
-import { bnOrZero } from 'lib/bignumber/bignumber'
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { MarketData } from '@shapeshiftoss/types'
+import type { BigNumber } from 'lib/bignumber/bignumber'
+import { bn, bnOrZero, convertPrecision } from 'lib/bignumber/bignumber'
+import type { ProtocolFee } from 'lib/swapper/api'
 
 export const convertBasisPointsToDecimalPercentage = (basisPoints: string) =>
   bnOrZero(basisPoints).div(10000)
 
 export const convertBasisPointsToPercentage = (basisPoints: string) =>
   bnOrZero(basisPoints).div(100)
+
+type SumProtocolFeesToDenomArgs = {
+  cryptoMarketDataById: Partial<Record<AssetId, Pick<MarketData, 'price'>>>
+  outputAssetPriceUsd: string
+  outputExponent: number
+  protocolFees: Record<AssetId, ProtocolFee>
+}
+
+export const sumProtocolFeesToDenom = ({
+  cryptoMarketDataById,
+  outputAssetPriceUsd,
+  outputExponent,
+  protocolFees,
+}: SumProtocolFeesToDenomArgs): string => {
+  return Object.entries(protocolFees)
+    .reduce((acc: BigNumber, [assetId, protocolFee]: [AssetId, ProtocolFee]) => {
+      const inputExponent = protocolFee.asset.precision
+      const priceUsd = cryptoMarketDataById[assetId]?.price
+
+      if (!inputExponent || !priceUsd) return acc
+
+      const rebasedAmountCryptoBaseUnit = convertPrecision({
```

Comment:
Nitpick but rebase has a very specific "rebasing token" (i.e FOXy/OHM forks) meaning around the codebase, if we're able to come up with a better naming for this

Reply:
renamed to `convertedPrecisionAmountCryptoBaseUnit` :)